providers/dynamodb/shedlock-provider-dynamodb/src/main/java/net/javacrumbs/shedlock/provider/dynamodb/DynamodbLockProvider.java

```diff
@@ -0,0 +1,306 @@
+     *                               determine the required throughput for dynamo db. Use this as tuning parameter
+     *                               to optimize the cost and avoid any throttling by dynamo db. Higher value
+     *                               will increase the cost and also increase throughput of dynamo db
+     */
+    public DynamodbLockProvider(AmazonDynamoDB amazonDynamoDB, String tableName, Duration minLockTime, long concurrentProcessCount) {
+        this(new DynamodbAccessor(amazonDynamoDB, tableName, minLockTime, concurrentProcessCount));
+    }
+
+    DynamodbLockProvider(DynamodbAccessor accessor) {
+        super(accessor);
+    }
+
+    public static class DynamodbAccessor extends AbstractStorageAccessor {
+        private boolean IS_TABLE_EXIST = false;
+        private final AmazonDynamoDB amazonDynamoDB;
+        private final String tableName;
+        private final String hostname;
+        private final Duration minLockTime;
+        private final long concurrentProcessCount;
+
+        DynamodbAccessor(AmazonDynamoDB amazonDynamoDB, String tableName, Duration minLockTime, long concurrentProcessCount) {
+            this.amazonDynamoDB = amazonDynamoDB;
+            this.tableName = tableName;
+            this.hostname = getHostname();
+            this.minLockTime = minLockTime;
+            this.concurrentProcessCount = concurrentProcessCount;
+            try {
+                createTableIfNotExists();
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        @Override
+        public boolean insertRecord(LockConfiguration lockConfiguration) {
+
+            DynamoDB dynamoDB = new DynamoDB(amazonDynamoDB);
+            Table table = dynamoDB.getTable(tableName);
+            UpdateItemSpec spec = new UpdateItemSpec()
+                    .withPrimaryKey(NAME, lockConfiguration.getName())
+                    .withUpdateExpression(
+                            String.format("set %s=:%s, %s=:%s, %s=:%s",
+                                    LOCKED_BY, LOCKED_BY, LOCK_UNTIL, LOCK_UNTIL, LOCKED_AT, LOCKED_AT))
+                    .withValueMap(
+                            new ValueMap()
+                                    .withString(":" + LOCKED_BY, lockConfiguration.getName())
+                                    .withLong(":" + LOCK_UNTIL, lockConfiguration.getLockAtMostUntil().toEpochMilli())
+                                    .withLong(":" + LOCKED_AT, Instant.now().toEpochMilli()))
+                    .withReturnValues(ReturnValue.ALL_OLD);
+
+            UpdateItemOutcome updateItemOutcome = table.updateItem(spec);
+            return (updateItemOutcome.getItem() == null);
+        }
+
+        private Map<String, AttributeValue> findLockRecord(LockConfiguration lockConfiguration) {
+            DynamoDB dynamoDB = new DynamoDB(amazonDynamoDB);
+            Table table = dynamoDB.getTable(tableName);
```

Comment:
getTable() can be a method (or even a field)

Reply:
Having table as a field would eliminate the need for creating the object again and again.